### ISSUE TITLE
Feature Request: Option to Disable the `"reported by reviewdog 🐶"` Prefix in Review Messages

### ISSUE THREAD

**Issue by mgaitan** [reviewdog/reviewdog#1662]
Hello reviewdog team!

First off, hats off to you for creating such a great tool! 🐶 It's been a fantastic addition to our CI workflow, sniffing out issues with the keen nose of a well-trained canine.

Now, onto the little flea in our codebase – We've noticed that every review comment comes with a "reported by reviewdog 🐶" prefix. While I appreciate the enthusiasm, it's a bit like having a "Commited using git" on every commit message. Cute, but you know, after the 104th message, the charm starts to wear off.

It's not that we're not proud of our four-legged friend; on the contrary, we've already given it a good scratch behind the ears and a place of honor in our workflow. But every time it barks out a report, the prefix feels a tad redundant – we know who's a good boy!

So, here's a frisbee to fetch – would it be possible to add a configuration option, either via the CLI and/or an environment variable, that allows us to disable the prefix message? This way, we can keep the comments clean and to the point, like a neatly groomed poodle.

Looking forward to your thoughts on this, and keep up the great work!

Best, 
Martin.

**Comment by Okabe-Junya** [comment]
Hello folks!!

For team, Thank you for the wonderful software !!
For mgaitan, great suggestion!!

I also agree with this proposal. Although I like seeing dogs, having every comment include "reported by reviewdog 🐶" might be a bit too much.

I have created a PR #1667 to fix this issue. PTAL :)

**Comment by haya14busa** [comment]
Actually, I'm surprised that I haven't received this feedback in the 7 years since reviewdog was created. Perhaps most users are happy with the comments from :dog:. The comments are designed to be minimally intrusive, without introducing any new lines and with a small font size to avoid being overly distracting.

I'm not strictly opposed to changing this, but I don't want to make any hasty decisions without careful consideration.

For example, by searching ["reported by reviewdog"](https://github.com/search?q=%22reported+by+reviewdog%22&type=pullrequests), you can find extensive use of reviewdog in OSS projects. This data could be potentially valuable. While there are currently no tools for analyzing this data, it is conceivable that such tools could be developed to leverage this information effectively.

**Comment by mgaitan** [comment]
@haya14busa in order to find the use "reviewdog in OSS projects" at least in github actions you can find directly in the workflows 

https://github.com/search?q=reviewdog++path%3A.github%2Fworkflows%2F+language%3AYAML&type=Code&ref=advsearch&l=YAML&l= 

I know it's a not a big deal, but this would be more precise as it avoids to compute potential comments like *"why this was not reported by reviewdog? Oh, we are not using it yet!*"

**Comment by haya14busa** [comment]
Comments with "reported by reviewdog" are  not only useful for finding reviewdog usage, but also useful for analysing integrated tools data.

**Comment by tvbeek** [comment]
To be honest I prefer to have it there. It is useful to see what tool reports a message.

I use it on GitLab and have a special user that place this report but is used for multiple tools. That means that the message makes clear what reports the message.

Maybe it is an idea to have it on by default and give to option to remove it from configuration?

**Comment by mgaitan** [comment]
> Maybe it is an idea to have it on by default and give to option to remove it from configuration?

that exactly is my proposal

**Comment by tvbeek** [comment]
> > Maybe it is an idea to have it on by default and give to option to remove it from configuration?
> 
> that exactly is my proposal

Yes, sorry I did read the issue, found it a good idea and then did read the linked PR where it is totally removed. That is where I mixed the 2 options. And wanted to explain why it is useful to have it visible (at least for me).

Your proposal sounds good for me 👍

**Comment by Okabe-Junya** [comment]
I think flags or configs are a very good idea too :) 

However, if it becomes possible to freely configure, wouldn't searches for comments with "reported by reviewdog" lose their value? (We would no longer be able to search through all OSS using reviewdog.)

Meaning, we will search projects through the `.github/workflows` files like thie.

> https://github.com/search?q=reviewdog++path%3A.github%2Fworkflows%2F+language%3AYAML&type=Code&ref=advsearch&l=YAML&l=

**Comment by tvbeek** [comment]
> I think flags or configs are a very good idea too :)
> 
> However, if it becomes possible to freely configure, wouldn't searches for comments with "reported by reviewdog" lose their value? (We would no longer be able to search through all OSS using reviewdog.)
> 
> Meaning, we will search projects through the `.github/workflows` files like thie.
> 
> > https://github.com/search?q=reviewdog++path%3A.github%2Fworkflows%2F+language%3AYAML&type=Code&ref=advsearch&l=YAML&l=

But that is only on GitHub not on other providers. And with a default I suspect that you will find between 90% and 99% of the projects that use it.

**Comment by Okabe-Junya** [comment]
> you will find between 90% and 99% of the projects that use it.

Of course, many projects can be found. However, if it's not 100%, there are more reliable methods of searching (although I'm not familiar with GitLab or other providers).

Meaning, if we adopt the idea of configuring with configs or flags (which I also agree), I will no longer use "reported by reviewdog" for data analysis or searching OSS like this

> Comments with "reported by reviewdog" are not only useful for finding reviewdog usage but also useful for analyzing integrated tools data.

**Comment by nogweii** [comment]
It would be more complicated, but could it instead mention the "reported by reviewdog" on only the first review comment? So it's still there, at least once in each MR, but isn't overly redundant by being in every single message.
